grpc/app_conn.go

```diff
@@ -34,15 +34,15 @@ type AppConn struct {
 // establishing a connection to App will continue to occur, however, in a background Goroutine. These attempts will continue until a
 // connection is made. If `cloud` is nil, an `AppConn` with a nil underlying connection will return, and the background dialer will not
 // start.
-func NewAppConn(ctx context.Context, appAddress, secret, id string, logger logging.Logger) (rpc.ClientConn, error) {
+func NewAppConn(ctx context.Context, appAddress, secret, id, apiKeyValue, apiKeyID string, logger logging.Logger) (rpc.ClientConn, error) {
```

Comment:
if at all possible, I'd like to avoid adding fields to a function where it'll be mutually exclusive with some of the other fields (in this case, half the time we'll only use secret and id, and the other half we'll only use api key & value). don't have great solutions, but either creating a second function `NewAppConnWithApiKey` or changing the args to take in some kind of secret/key interface would be preferred.